app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -286,7 +286,57 @@ public void onClick(View view) {
                     startActivity(intent);
                 } else {
                     final Intent intent = new Intent(Intent.ACTION_VIEW,
-                            Uri.parse("market://details?id=" + Browsers.KnownBrowser.FIREFOX));
+                            Uri.parse("market://details?id=" + Browsers.KnownBrowser.FIREFOX.packageName));
+                    startActivity(intent);
+                }
+
+                break;
+            }
+
+            case R.id.open_firefox_beta: {
```

Comment:
This here and in `BrowserMenuAdapter` is a lot of duplicated code. Can we hide this functionality in the `Browsers` class? We might just need a `ActivityInfo getFirefoxBrandedBrowser()` method which we can use to determine whether we want to show a menu item and also for opening the browser. Almost like the default browser item, but just for firefox branded browsers.